src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/ResourceManagerCollectionSetup.cs

```diff
@@ -0,0 +1,253 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.CosmosDB;
+using Azure.ResourceManager.CosmosDB.Models;
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.CosmosDb.Core.Configs;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.StoredProcedures;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.Versioning;
+using Polly;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Initialization.Features.Storage;
+
+/// <summary>
+/// Sets up the Cosmos FHIR Database using the ResourceManager SDK and ManagedIdentity.
+/// Requires the MI connection to have 'Cosmos DB Operator' permissions.
+/// </summary>
+public class ResourceManagerCollectionSetup : ICollectionSetup
+{
+    private readonly ILogger<ResourceManagerCollectionSetup> _logger;
+    private readonly ArmClient _armClient;
+    private readonly IOptionsMonitor<CosmosCollectionConfiguration> _cosmosCollectionConfiguration;
+    private readonly CosmosDataStoreConfiguration _cosmosDataStoreConfiguration;
+    private readonly IEnumerable<IStoredProcedureMetadata> _storeProceduresMetadata;
+    private readonly ResourceIdentifier _resourceIdentifier;
+    private CosmosDBSqlDatabaseResource _database;
+    private AzureLocation? _location;
+    private readonly CosmosDBAccountResource _account;
+
+    public ResourceManagerCollectionSetup(
+        IOptionsMonitor<CosmosCollectionConfiguration> cosmosCollectionConfiguration,
+        CosmosDataStoreConfiguration cosmosDataStoreConfiguration,
+        IConfiguration genericConfiguration,
+        IEnumerable<IStoredProcedureMetadata> storedProcedures,
+        ILogger<ResourceManagerCollectionSetup> logger)
+    {
+        EnsureArg.IsNotNull(storedProcedures, nameof(storedProcedures));
+
+        _cosmosCollectionConfiguration = EnsureArg.IsNotNull(cosmosCollectionConfiguration, nameof(cosmosCollectionConfiguration));
+        _cosmosDataStoreConfiguration = EnsureArg.IsNotNull(cosmosDataStoreConfiguration, nameof(cosmosDataStoreConfiguration));
+        _storeProceduresMetadata = EnsureArg.IsNotNull(storedProcedures, nameof(storedProcedures));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _armClient = new ArmClient(new DefaultAzureCredential());
+        var dataStoreResourceId = genericConfiguration.GetValue("FhirServer:ResourceManager:DataStoreResourceId", string.Empty);
+        _resourceIdentifier = ResourceIdentifier.Parse(dataStoreResourceId);
```

Comment:
Do we need to pass dataStoreResourceId for AKS and SF?

I think SF and AKS would not be using the functionality of this class correct as RP would be doing the work for managed services. If that's true then registering ResourceManagerCollectionSetup here fails  https://github.com/microsoft/fhir-server/pull/4026/files#r1742470746

Reply:
Yes if we use this provider, but we do also have a Noop one I believe